src/graph/query_graph.c

```diff
@@ -376,6 +376,39 @@ uint QueryGraph_EdgeCount(const QueryGraph *qg) {
 	return array_len(qg->edges);
 }
 
+GrB_Matrix QueryGraph_MatrixRepresentation(const QueryGraph *qg) {
+	assert(qg);
+
+	// Make a clone of the given graph as we're about to modify it.
+	QueryGraph *qg_clone = QueryGraph_Clone(qg);
```

Comment:
Instead of cloning and modifying the QueryGraph input, we could populate a rax with Node ID keys and `i` values. That might actually be a more expensive approach, though, depending on the number of `dest` lookups. (No recommendation here.)